includes/REST_API/Stories_Base_Controller.php

```diff
@@ -58,8 +68,151 @@ class Stories_Base_Controller extends WP_REST_Posts_Controller {
+					],
+				],
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'get_lock' ],
+					'permission_callback' => [ $this, 'update_item_permissions_check' ],
+					'args'                => $lock_args,
+				],
+				[
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => [ $this, 'update_lock' ],
+					'permission_callback' => [ $this, 'update_item_permissions_check' ],
+					'args'                => $lock_args,
+				],
+				[
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => [ $this, 'delete_lock' ],
+					'permission_callback' => [ $this, 'update_item_permissions_check' ],
+					'args'                => $lock_args,
+				],
+			]
+		);
+	}
+
+	/**
+	 * Get post lock
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success.
+	 */
+	public function get_lock( $request ) {
+		return $this->prepare_lock_for_response( $request );
+	}
+
+	/**
+	 * Update post lock
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success.
+	 */
+	public function update_lock( $request ) {
+		if ( ! function_exists( 'wp_set_post_lock' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/post.php';
+		}
+
+		wp_set_post_lock( $request['id'] );
+
+		return $this->prepare_lock_for_response( $request );
+	}
+
+	/**
+	 * Delete post lock
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response Response object on success.
+	 */
+	public function delete_lock( $request ) {
+		$previous = $this->prepare_lock_for_response( $request );
+		$result   = delete_post_meta( $request['id'], '_edit_lock' );
+		$data     = [];
+		if ( ! is_wp_error( $previous ) ) {
+			$data = $previous->get_data();
+		}
+		$response = new WP_REST_Response();
+		$response->set_data(
+			[
+				'deleted'  => $result,
+				'previous' => $data,
+			]
+		);
+
+		return $response;
+	}
+
+	/**
+	 * Prepares a single lock output for response.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object.
+	 */
+	public function prepare_lock_for_response( $request ) {
+		$lock = get_post_meta( $request['id'], '_edit_lock', true );
+
+		$data  = [
+			'locked' => false,
+		];
+		$links = [];
+
+		if ( $lock ) {
+			$lock                 = explode( ':', $lock );
+			list ( $time, $user ) = $lock;
+
+			/** This filter is documented in wp-admin/includes/ajax-actions.php */
+			$time_window = apply_filters( 'wp_check_post_lock_window', 150 );
+
+			if ( $time && $time > time() - $time_window ) {
+				$data = [
+					'locked' => true,
+					'time'   => $time,
+					'user'   => (int) $user,
+				];
+
+				$links['author'] = [
+					'href'       => rest_url( 'wp/v2/users/' . $user ),
+					'embeddable' => true,
+				];
+			}
+		}
```

Comment:
Since `$time` is not really used on the client and all we need is the user (if locked), we should be able to just use `wp_check_post_lock()` here. It would greatly simplify the code here:


```php
$user_id = wp_check_post_lock();

$data  = [
	'locked' => (bool) $user_id,
    'user'   => (int) $user_id,
];
$links = [];

$response = rest_ensure_response( $data );
if ( ! is_wp_error( $response ) && $user_id ) {
	$response->add_links(
		[
			'author' => [
				'href'       => rest_url( 'wp/v2/users/' . $user_id ),
				'embeddable' => true,
			],
		]
	);
}

return $response;
```

We don't even need the `locked` array key as it's redundant. If there is a user ID you know it's locked. No user ID, no lock.

Reply:
I was trying to build a more generic API here, so adding time makes sense for other implementations here. I found it useful for debugging and I think it makes perfect sense here. Trying not to build an REST API just for our use case. 

I also return time and user as this what [wp_set_post_lock](https://developer.wordpress.org/reference/functions/wp_set_post_lock/), so keeps the output generate on gets and set. Core is messy and weird on this one. 

I personally like having fields in apis that makes it clear the state. You don't need to post status, you can get that is published by the date but it there. A field like this makes it a lot more human readable.